src/core/column/minmax.h

```diff
@@ -41,29 +42,32 @@ class MinMax_ColumnImpl : public ReduceUnary_ColumnImpl<T, IS_GROUPED> {
         *out = value;
         return is_valid;
       } else {
-          T minmax = MIN ? std::numeric_limits<T>::min()
-                         : std::numeric_limits<T>::max();
-          bool isna = true;
+          T minmax = MIN ? (std::numeric_limits<T>::has_infinity ? std::numeric_limits<T>::infinity() 
+                                                                 : std::numeric_limits<T>::max()
+                           )
+                         : (std::numeric_limits<T>::has_infinity ? std::numeric_limits<T>::lowest() 
+                                                                 : std::numeric_limits<T>::min()
+                           );  
```

Comment:
I'm not sure why you need a check for `has_infinity` here. We should just take the algo currently implemented, i.e.
- for min: https://github.com/h2oai/datatable/blob/main/src/core/expr/head_reduce_unary.cc#L531-L548
- for max: https://github.com/h2oai/datatable/blob/main/src/core/expr/head_reduce_unary.cc#L483-L500

and make it a single implementation based on the value of `MIN`. 

Btw, it seems that your previous implementation (where you had `std::numeric_limits<T>::max()` and `std::numeric_limits<T>::min()` vice versa) passed all the tests, because the default value for `minmax` doesn't really matter even in our current algo. That's due to the check `|| min_isna` (`|| max_isna`), which actually allows the `minmax` update on the first element not depending on its initial value.